### PR TITLE
Update android SDK and Java version [3.6]

### PR DESCRIPTION
**Pull request by avirahmandev**
- Update Java version from 11 to 17
- Update target SDK from 33 to 34 according to Google's requirements for publishing on the Play Store

### CODE REVIEW

**Comment by avirahmandev** [comment]
What does the error mean? This is the first PR I've done, I don't know much about it yet

**Comment by AThousandShips** [comment]
They are unrelated to your additions, they are errors in the spell check system that haven't been fixed in 3.x yet, see:
* https://github.com/godotengine/godot-docs/pull/9414

Though there is one that is a proper typo, but regardless it's unrelated to your PR

File: tutorials/export/exporting_for_android.rst [hunk 1]
```diff
@@ -41,7 +41,7 @@ Download and install the Android SDK.
 
 ::
 
-    sdkmanager --sdk_root=<android_sdk_path> "platform-tools" "build-tools;30.0.3" "platforms;android-31" "cmdline-tools;latest" "cmake;3.10.2.4988404" "ndk;21.4.7075529"
+    sdkmanager --sdk_root=<android_sdk_path> "platform-tools" "build-tools;30.0.3" "platforms;android-34" "cmdline-tools;latest" "cmake;3.10.2.4988404" "ndk;21.4.7075529"
```
Comment (m4gr3d) [inline tutorials/export/exporting_for_android.rst:44]: Can you update the `build-tools` and the `ndk` versions as well, the `ndk` version is listed above but for some reasons it wasn't updated in this command.

**Review by m4gr3d — CHANGES_REQUESTED** [review]

File: tutorials/export/exporting_for_android.rst (hunk 1 above)
Comment (avirahmandev) [reply tutorials/export/exporting_for_android.rst]: Sure!

File: tutorials/export/exporting_for_android.rst [hunk 2]
```diff
@@ -29,19 +29,19 @@ Download and install the Android SDK.
...
 
     - Android SDK Platform-Tools version 30.0.5 or later
-    - Android SDK Build-Tools version 30.0.3
-    - Android SDK Platform 33
+    - Android SDK Build-Tools version 34.0.0
+    - Android SDK Platform 34
     - Android SDK Command-line Tools (latest)
     - CMake version 3.10.2.4988404
-    - NDK version r23c (23.2.8568313)
+    - NDK version r26c (26.2.11394342)
```
Comment (m4gr3d) [inline tutorials/export/exporting_for_android.rst:36]: For the `ndk` version, we want to remain at `23.2.8568313` for the time being (the newer versions modify the build process which also requires us to update Godot Android build process, so we're deferring the task for now).

**Review by m4gr3d — CHANGES_REQUESTED** [review]

File: tutorials/export/exporting_for_android.rst (hunk 2 above)
Comment (avirahmandev) [reply tutorials/export/exporting_for_android.rst:36]: Sorry, I didn't realize that the version listed in the list was not the same as the one in the command line, I thought it needed to be updated to the newer.

**Review by m4gr3d — APPROVED** [review]
Looks good, thanks for the update!

**Review by skyace65 — APPROVED** [review]

**Comment by skyace65** [comment]
Build issue was fixed so I just rebased your PR so it would build correctly. Congrats on your first merged PR!

Some advice for the future, when making a PR don't use your master or 3.6 branch, you'll want to fork them, not doing that makes updating your fork of the docs very hard. In addition please keep PRs to 1 commit, I did a squash and merge this time but keep that in mind for the future. If you haven't used git before we have a guide [here](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html).

**Comment by avirahmandev** [comment]
> Build issue was fixed so I just rebased your PR so it would build correctly. Congrats on your first merged PR!
> 
> Some advice for the future, when making a PR don't use your master or 3.6 branch, you'll want to fork them, not doing that makes updating your fork of the docs very hard. In addition please keep PRs to 1 commit, I did a squash and merge this time but keep that in mind for the future. If you haven't used git before we have a guide [here](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html).

Thanks! for your merge & advice, very insightful, learned a lot here to start contributing to godot!

**Pull request merged**